src/include/encoding.h

```diff
@@ -246,6 +246,23 @@ inline void encode(const char *s, bufferlist& bl)
   encode(std::string_view(s, strlen(s)), bl);
 }
 
+// opaque byte vectors
+inline void encode(std::vector<uint8_t>& v, bufferlist& bl)
+{
+  uint32_t len = v.size();
+  encode(len, bl);
+  if (len)
+    bl.append((char *)v.data(), len);
+}
+
+inline void decode(std::vector<uint8_t>& v, bufferlist::const_iterator& p)
+{
+  uint32_t len;
+
+  decode(len, p);
+  v.resize(len);
+  p.copy(len, (char *)v.data());
+}
```

Comment:
BTW, what can we benefit from these two opaque routines ? Except it will be faster than the old template routine ? And is this the purpose of these ?

Reply:
The templated routines didn't cover these objects. Without defining encode/decode routines for this, it failed to compile:
```
/home/jlayton/git/ceph/src/mds/CInode.cc:1680:32: error: no matching  function for call to ‘decode(std::vector<unsigned char>&)’
```

Reply:
Okay, I thought the https://github.com/ceph/ceph/blob/master/src/include/encoding.h#L877 should work.

Reply:
It may. Perhaps I just needed to pass the "features" parameter? I can try it again for the next respin.

Reply:
It turns out not to be terribly helpful, as the features parameter is only for the encode side, whereas the decode side doesn't take that parameter. I'm going to leave these routines in place for now, but I'm open to dropping them and using existing code if someone can suggest how to do it.